app/models/spree/payment/processing.rb

```diff
@@ -23,6 +23,7 @@ def process_offline!
       def authorize!(return_url = nil)
         started_processing!
         gateway_action(source, :authorize, :pend, return_url: return_url)
+        raise Spree::Core::GatewayError, I18n.t('authorization_failure') unless pending?
```

Comment:
that's something that only applies to StripeSCA, so we should move it deeper into `Gateway::StripeSCA`.

Reply:
If we can't progress to `pending` it might be because a model validation failed. In that case, we might want to see that in Bugsnag, and it has nothing to do with Gateway errors. It feels like we're missing concerns.

Reply:
@andrewpbrett as discussed, a7e1b7e67bde000fe3beb0941e17a051bc576e3a makes things clear while keeping the behaviour.